framework/src/main/java/org/checkerframework/framework/util/JavacParseUtil.java

```diff
@@ -0,0 +1,113 @@
+import com.sun.source.tree.VariableTree;
+import com.sun.source.util.JavacTask;
+import com.sun.source.util.TreeScanner;
+import com.sun.tools.javac.api.JavacTool;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Collections;
+import javax.tools.JavaCompiler;
+import javax.tools.JavaFileObject;
+import javax.tools.SimpleJavaFileObject;
+import javax.tools.StandardJavaFileManager;
+
+/**
+ * Utility class that provides methods for parsing Java expressions using the {@code javac} compiler
+ * API.
+ *
+ * <p>This utility uses {@link JavacTask} and the standard {@code com.sun.source.tree} API to parse
+ * expressions.
+ */
+public class JavacParseUtil {
+
+  /**
+   * Parses the given Java expression string and returns it as a {@link ExpressionTree} using the
+   * {@code javac} compiler API.
+   *
+   * <p>The method works by embedding the expression in a dummy class and variable declaration, then
+   * parsing the resulting source to extract the expression tree.
+   *
+   * <p>For example, the input {@code "1 + 2"} is transformed into:
+   *
+   * <pre>{@code
+   * class Dummy { Object expression = 1 + 2; }
+   * }</pre>
+   *
+   * The initializer of the {@code expression} field is then extracted and returned.
+   *
+   * @param expressionSource the string representation of a Java expression (e.g., "foo.bar()", "1 +
+   *     2")
+   * @return the parsed {@link ExpressionTree}
+   * @throws RuntimeException if parsing fails or the expression cannot be found in the AST
+   */
+  public static ExpressionTree parseExpression(String expressionSource) {
+    // Embed the expression inside a dummy class and variable declaration
+    String dummySource = "class Dummy { Object expression = " + expressionSource + "; }";
+
+    // Obtain the system Java compiler
+    JavaCompiler compiler = JavacTool.create();
+
+    // Create an in-memory Java file from the dummy source code
+    JavaFileObject fileObject =
+        new SimpleJavaFileObject(URI.create("string:///Dummy.java"), JavaFileObject.Kind.SOURCE) {
+          @Override
+          public CharSequence getCharContent(boolean ignoreEncodingErrors) {
+            return dummySource;
+          }
+        };
+
+    try {
+      // Prepare the file manager and task
+      StandardJavaFileManager fileManager = compiler.getStandardFileManager(null, null, null);
+      JavacTask task =
+          (JavacTask)
+              compiler.getTask(
+                  null,
+                  fileManager,
+                  null,
+                  Collections.emptyList(),
+                  null,
+                  Collections.singletonList(fileObject));
+
+      // Parse the source and extract the CompilationUnit
+      Iterable<? extends CompilationUnitTree> trees = task.parse();
+      CompilationUnitTree cu = trees.iterator().next();
+
+      // Use a TreeScanner to extract the expression from the variable initializer
+      ExpressionExtractor extractor = new ExpressionExtractor();
```

Comment:
There is exactly one expression, so there is no need to use a visitor.  (And, the use of a visitor led me to wonder what would happen if there were more than one expression.)  The code will be simpler and shorter if you just access the expression by accessing the CompilationUnitTree's first class, then the class's first member, then that declaration's initializer expression.  I agree that the 4 or 5 statements to obtain that are a bit gross, but I think simplicity is a greater virtue in this case.